drivers/sensor/tmd2620/tmd2620.c

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2022 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ams_tmd2620
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <string.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/gpio.h>
+#include <stdio.h>
+
+#include "tmd2620.h"
+
+LOG_MODULE_REGISTER(TMD2620, CONFIG_SENSOR_LOG_LEVEL);
+
+static void tmd2620_gpio_callback(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+
+	LOG_DBG("Interrupt Callback was called");
+
+	struct tmd2620_data *data = CONTAINER_OF(cb, struct tmd2620_data, gpio_cb);
+
+	tmd2620_setup_int(data->dev->config, false);
+
+#ifdef CONFIG_TMD2620_TRIGGER
+	k_work_submit(&data->work);
+#else
+	k_sem_give(&data->data_sem);
+#endif
+}
+
+static int tmd2620_configure_interrupt(const struct device *dev)
+{
+	struct tmd2620_data *data = dev->data;
+	const struct tmd2620_config *config = dev->config;
+
+	LOG_DBG("Configuring Interrupt.");
+
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("Interrupt GPIO device not ready");
+		return -EIO;
+	}
+
+	gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT | config->int_gpio.dt_flags);
+
+	gpio_init_callback(&data->gpio_cb, tmd2620_gpio_callback, BIT(config->int_gpio.pin));
+
+	if (gpio_add_callback(config->int_gpio.port, &data->gpio_cb) < 0) {
+		LOG_ERR("Failed to set GPIO callback");
+		return -EIO;
```

Comment:
Don't use `-EIO` everywhere, please propagate the rc from functions instead.

Reply:
Do I understand you correctly, I should return the return value of the error-producing function?
In this example the return value of `gpio_add_callback(config->int_gpio.port, &data->gpio_cb)` ?

Reply:
<pre>
ret = gpio_add_callback(config->int_gpio.port, &data->gpio_cb);
if (ret < 0)  {
	LOG_ERR("Failed to set GPIO callback");
	return ret;
}
</pre>

This way we would know the actual return code, instead of the hardcoded `EIO`;

Reply:
Okay, I understand. I'll change that